service/routes.py

```diff
@@ -44,3 +49,97 @@ def index():
 ######################################################################
 
 # Todo: Place your REST API code here ...
+######################################################################
+# LIST ALL PROMOTIONS
+######################################################################
+@app.route("/promotions", methods=["GET"])
+def list_promotions():
+    """Returns all of the Promotions"""
+    app.logger.info("Request for promotion list")
+
+    promotions = []
+
+    # Parse any arguments from the query string
+    title = request.args.get("title")
```

Comment:
Do we need to support get by title here? Shouldn't it only be get all promotions?

Reply:
I think we need to support get by title, because we design the class method find_by_title() in models.py.
![WechatIMG524](https://github.com/user-attachments/assets/8c9fa2b6-277d-4af1-a334-5c9ee22c4e06)

Reply:
Oh I see, but we might want to move it to a different route/function as this is for listing all promotions.

Reply:
Good point! That will be better. I'll change it. BTW, do we need to make the git commit clean, like I rebase to previous commit to update codes and reserve the original commit logs, so that there will not be too many new commits to handle some tiny problems like typo or code style. It doesn't matter with the project, just make the commit log more logical.

Reply:
I don't think we need to worry about number of commits, as long as the changes are there.